### PR TITLE
Added script support to status changes

### PR DESCRIPTION
**Pull request by Lemongrass3110**
* **Addressed Issue(s)**: None

* **Server Mode**: Both

* **Description of Pull Request**: 
Adds the ability to script status changes for players

### CODE REVIEW

File: src/map/status.cpp [hunk 1]
```diff
@@ -10171,6 +10181,11 @@ int status_change_start(struct block_list* src, struct block_list* bl,enum sc_ty
 	if( !sc )
 		return 0; // Unable to receive status changes
 
+	// Scripted status changes only work for players for the time being
+	if( scdb->script != nullptr && bl->type != BL_PC ){
```
Comment (Atemo) [inline src/map/status.cpp:10185]: A debug/warning would be welcome to track potential mistake

Comment (Lemongrass3110) [inline src/map/status.cpp:10185]:
```suggestion
	if( scdb->script != nullptr && bl->type != BL_PC ){
		ShowError( "status_change_start: Trying to start the scripted status change %d on a non player.\n", type );
```
Like this?

Comment (Atemo) [reply src/map/status.cpp:10185]: Yes, maybe "failed" would be more explicit than "trying" tho

**Review by aleos89 — APPROVED** [review]
Looks good! Just missing doc/status.txt addition for the new parameter!

**Review by Atemo — APPROVED** [review]

**Review by aleos89 — APPROVED** [review]

**Pull request merged**